Include/Beman/InplaceVector/InplaceVector.hpp

```diff
@@ -0,0 +1,750 @@
+    return static_cast<size_type>(this->size_);
+  }
+  constexpr bool empty() const noexcept { return this->size_ == 0; }
+
+  // [containers.sequences.inplace.vector.data], data access
+  constexpr T *data() noexcept {
+    return std::launder(reinterpret_cast<T *>(this->elems));
+  }
+  constexpr const T *data() const noexcept {
+    return std::launder(reinterpret_cast<const T *>(this->elems));
+  }
+
+  // [containers.sequences.inplace.vector.iterators] iterators
+  iterator begin() noexcept {
+    return std::launder(reinterpret_cast<T *>(this->elems));
+  }
+  const_iterator begin() const noexcept {
+    return std::launder(reinterpret_cast<const T *>(this->elems));
+  }
+
+  iterator end() noexcept {
+    return std::launder(reinterpret_cast<T *>(this->elems) + this->size());
+  }
+
+  const_iterator end() const noexcept {
+    return std::launder(reinterpret_cast<const T *>(this->elems) +
+                        this->size());
+  }
+
+  // [containers.sequences.inplace.vector.modifiers], modifiers
+  static constexpr void uninitialized_value_construct(iterator first,
+                                                      iterator last) noexcept {
+    std::fill(first, last, T());
+  }
+
+  static constexpr void uninitialized_fill(iterator first, iterator last,
+                                           const T &value) noexcept {
+    std::fill(first, last, value);
+  }
+
+  template <typename Iter>
+  static constexpr void uninitialized_copy(Iter first, Iter last,
+                                           iterator dest) noexcept {
+    std::copy(first, last, dest);
+  }
+
+  template <typename Iter>
+  constexpr void uninitialized_move(Iter first, Iter last,
+                                    iterator dest) noexcept {
+    std::copy(first, last, dest);
+  }
+};
+
+template <class T, std::size_t Capacity>
+class inplace_vector : public inplace_vector_base<T, Capacity> {
+private:
+  using base = inplace_vector_base<T, Capacity>;
+
+public:
+  using value_type = T;
+  using pointer = T *;
+  using const_pointer = const T *;
+  using reference = value_type &;
+  using const_reference = const value_type &;
+  using size_type = std::size_t;
+  using difference_type = std::ptrdiff_t;
+  using iterator = pointer;
+  using const_iterator = const_pointer;
+  using reverse_iterator = std::reverse_iterator<iterator>;
+  using const_reverse_iterator = std::reverse_iterator<const_iterator>;
+
+  // [containers.sequences.inplace.vector.cons], construct/copy/destroy
+  constexpr inplace_vector() noexcept = default;
+  constexpr inplace_vector(const inplace_vector &) = default;
+  constexpr inplace_vector(inplace_vector &&) = default;
+  constexpr inplace_vector &operator=(const inplace_vector &) = default;
+  constexpr inplace_vector &operator=(inplace_vector &&) = default;
+
+  constexpr explicit inplace_vector(const size_type size) : base(size) {
+    if (size > 0) {
+      base::uninitialized_value_construct(this->begin(), this->end());
```

Comment:
`std:fill` will deal with already initialised(constructed) memory I think, if you track down what base(size) does you will end up with:
```cpp
  constexpr inplace_vector_destruct_base(const size_type size)
      : elems(), size_(size) {
 ```
 Here elems() does nothing because the buffer is already initialised to zero by `{}`
 ```cpp
   alignas(T) unsigned char elems[Capacity * sizeof(T)] = {};
 ```
  It basically does zero-initialisation(hence the `{}`).